src/docdb/tree/DocDBTriggersTreeItem.ts

```diff
@@ -0,0 +1,121 @@
+    }
+
+    public async createChildImpl(context: ICreateChildImplContext): Promise<DocDBTriggerTreeItem> {
+        const client = this.root.getCosmosClient();
+        const currTriggerList: AzExtTreeItem[] = await this.getCachedChildren(context);
+        const currTriggerNames: string[] = [];
+        for (const sp of currTriggerList) {
+            currTriggerNames.push(nonNullProp(sp, "id"));
+        }
+
+        const triggerID = (await context.ui.showInputBox({
+            prompt: "Enter a unique trigger ID",
+            stepName: 'createTrigger',
+            validateInput: (name: string) => this.validateTriggerName(name, currTriggerNames)
+        })).trim();
+
+        const triggerType = await getTriggerType(context);
+        const triggerOperation = await getTriggerOperation(context);
+
+        const body: TriggerDefinition = { id: triggerID, body: defaultTrigger, triggerType: triggerType, triggerOperation: triggerOperation };
+        context.showCreatingTreeItem(triggerID);
+        const response = await this.getContainerClient(client).scripts.triggers.create(body);
+
+        return this.initChild(nonNullProp(response, 'resource'));
+    }
+
+    public get id(): string {
+        return "$Triggers";
+    }
+
+    public get label(): string {
+        return "Triggers";
+    }
+
+    public get link(): string {
+        return this.parent.link;
+    }
+
+    public getIterator(client: CosmosClient, feedOptions: FeedOptions): QueryIterator<TriggerDefinition & Resource> {
+        return this.getContainerClient(client).scripts.triggers.readAll(feedOptions);
+    }
+
+    public getContainerClient(client: CosmosClient): Container {
+        return this.parent.getContainerClient(client);
+    }
+
+    private validateTriggerName(name: string, currTriggerNames: string[]): string | undefined {
+        if (name.length < 1 || name.length > 255) {
+            return localize("nameLength", "Name has to be between 1 and 255 chars long");
+        }
+
+        if (/[/\\?#&]/.test(name)) {
+            return localize("illegalChars", "Name contains illegal chars: /, \\, ?, #, &");
+        }
+        if (name[name.length - 1] === " ") {
+            return localize("endsWithSpace", "Name cannot end with a space.");
+        }
+        if (currTriggerNames.includes(name)) {
+            return localize('nameExists', 'Trigger "{0}" already exists.', name);
+        }
+
+        return undefined;
+    }
+}
+
+export async function getTriggerType(context: IActionContext): Promise<TriggerType> {
+    const triggerTypeOption = (await context.ui.showQuickPick<vscode.QuickPickItem>([
+        { label: "Pre" },
+        { label: "Post" }
+    ], {})).label as "Pre" | "Post";
+    return triggerTypeOption === "Pre" ? TriggerType.Pre : TriggerType.Post;
+}
+
+export async function getTriggerOperation(context: IActionContext): Promise<TriggerOperation> {
+    const triggerOperationOption = (await context.ui.showQuickPick<vscode.QuickPickItem>([
```

Comment:
nit: picks can be a little cleaner

`const triggerOperationOption = (await context.ui.showQuickPick<vscode.QuickPickItem>(
        Object.keys(TriggerOperation).map((key) => ({ label: key })),
        {})).label as "All" | "Create" | "Delete" | "Replace" | "Update";`

Reply:
Done.